pytorch_lightning/utilities/device_dtype_mixin.py

```diff
@@ -0,0 +1,11 @@
+from pytorch_lightning.utilities import rank_zero_deprecation
+
+rank_zero_deprecation(
```

Comment:
can we rather use something like [this](https://github.com/kaushikb11/pytorch-lightning/blob/bc/mixin/pytorch_lightning/metrics/utils.py#L32) (being used [here](https://github.com/kaushikb11/pytorch-lightning/blob/bc/mixin/pytorch_lightning/metrics/classification/accuracy.py#L23) for example)? This only raises the warning upon `__init__`